VungleAdapter/src/main/java/com/chartboost/mediation/vungleadapter/VungleAdapter.kt

```diff
@@ -664,153 +524,198 @@ class VungleAdapter : PartnerAdapter {
+                AdFormat.INTERSTITIAL -> loadVungleFullScreenAd {
+                    InterstitialAd(context, request.partnerPlacement, adConfig)
+                }
+                AdFormat.REWARDED -> loadVungleFullScreenAd {
+                    RewardedAd(context, request.partnerPlacement, adConfig)
+                }
+                else -> {
+                    if (request.format.key == "rewarded_interstitial") {
+                        loadVungleFullScreenAd {
+                            RewardedAd(context, request.partnerPlacement, adConfig)
+                        }
+                    } else {
+                        PartnerLogController.log(LOAD_FAILED)
                         resumeOnce(
                             Result.failure(
                                 ChartboostMediationAdException(
-                                    getChartboostMediationError(
-                                        exception
-                                    )
+                                    ChartboostMediationError.CM_LOAD_FAILURE_UNSUPPORTED_AD_FORMAT
                                 )
                             )
                         )
                     }
                 }
-            )
+            }
         }
     }
 
+
     /**
      * Attempt to show a Vungle fullscreen ad.
      *
      * @param partnerAd The [PartnerAd] object containing the Vungle ad to be shown.
-     * @param listener A [PartnerAdListener] to notify Chartboost Mediation of ad events.
+     *
+     * @return Result.success(PartnerAd) if the ad was successfully shown, Result.failure(Exception) otherwise.
      */
-    private suspend fun showFullscreenAd(
-        partnerAd: PartnerAd,
-        listener: PartnerAdListener?
+    private fun showFullscreenAd(
+        partnerAd: PartnerAd
     ): Result<PartnerAd> {
-        val adm = adms.remove(partnerAd.request.partnerPlacement)
-
-        return suspendCancellableCoroutine { continuation ->
-            fun resumeOnce(result: Result<PartnerAd>) {
-                if (continuation.isActive) {
-                    continuation.resume(result)
-                }
+        fun canPlay(canPlay: () -> Boolean, play: () -> Unit): Result<PartnerAd> {
+            return if (canPlay()) {
+                play()
+                Result.success(partnerAd)
+            } else {
+                PartnerLogController.log(SHOW_FAILED)
+                Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_SHOW_FAILURE_AD_NOT_READY))
             }
+        }
 
-            if (Vungle.canPlayAd(partnerAd.request.partnerPlacement, adm)) {
-                Vungle.playAd(
-                    partnerAd.request.partnerPlacement,
-                    adm,
-                    null,
-                    object : PlayAdCallback {
-                        override fun creativeId(creativeId: String) {
-                            // Ignored.
-                        }
+        onShowSuccess = {
```

Comment:
you don't need these lambdas if you aren't using them for anything. there's no difference if you in-line these when you create the listener.